src/activities/solar_system/Solar_system.qml

```diff
@@ -0,0 +1,321 @@
+/* GCompris - Solar_system.qml
+ *
+ * Copyright (C) 2018 Aman Kumar Gupta <gupta2140@gmail.com>
+ *
+ * Authors:
+ *    Aman Kumar Gupta <gupta2140@gmail.com>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.6
+import GCompris 1.0
+import "../../core"
+import "solar_system.js" as Activity
+
+ActivityBase {
+    id: activity
+
+    onStart: {
+        focus = true;
+    }
+    onStop: {}
+
+    pageComponent: Image {
+        id: background
+        anchors.fill: parent
+        fillMode: Image.PreserveAspectCrop
+        source: "qrc:/gcompris/src/activities/solar_system/resource/background.jpg"
+
+        property bool horizontalLayout: background.width > background.height
+
+        signal start
+        signal stop
+
+        Component.onCompleted: {
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property alias background: background
+            property alias bar: bar
+            property alias bonus: bonus
+            property alias planetsModel: planetsModel
+            property alias mainQuizScreen: mainQuizScreen
+            property alias dialogActivityConfig: dialogActivityConfig
+            property bool assessmentMode: false
+            property bool solarSystemVisible: true
+            property bool quizScreenVisible: false
+            property string temperatureHint
+            property string lengthOfYearHint
+        }
+
+        onStart: {
+            dialogActivityConfig.getInitialConfiguration()
+            Activity.start(items)
+        }
+
+        onStop: {
+            dialogActivityConfig.saveDatainConfiguration()
```

Comment:
you don't need to save the configuration here, it's already done when changing it